session/txnInfo/txnInfo.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2021 PingCAP, Inc.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package txnInfo
+
+import (
+	"time"
+
+	"github.com/pingcap/parser/mysql"
+	"github.com/pingcap/tidb/store/tikv/oracle"
+	"github.com/pingcap/tidb/types"
+)
+
+// TxnRunningState is the current state of a transaction
+type TxnRunningState = int32
+
+const (
+	// TxnRunningNormal means the transaction is running normally
+	TxnRunningNormal TxnRunningState = iota
+	// TxnLockWaiting means the transaction is blocked on a lock
+	TxnLockWaiting TxnRunningState = iota
+	// TxnCommitting means the transaction is (at least trying to) committing
+	TxnCommitting TxnRunningState = iota
+	// TxnRollingBack means the transaction is rolling back
+	TxnRollingBack TxnRunningState = iota
+)
+
+// TxnInfo is information about a running transaction
+// This is supposed to be the datasource of `TIDB_TRX` in infoschema
+type TxnInfo struct {
```

Comment:
We could add some examples query the new view in the PR description to make it look intuitive.